model/renderNativeWindow.h

```diff
@@ -24,12 +24,44 @@
 
 namespace minsky
 {
-  class RenderNativeWindow: public ecolab::CairoSurface
+  struct WindowInformation
+  {
+    unsigned long parentWindowId;
+    unsigned long childWindowId;
+    int childWidth;
+    int childHeight;
+    int offsetLeft;
+    int offsetTop;
+    Display*	display;
+    XWindowAttributes wAttr;
+    ecolab::cairo::SurfacePtr *childSurface; // TODO:: Review right way to store surface ptr
+
+    WindowInformation();
+    ~WindowInformation();
+    void initialize(unsigned long parentWin, int offsetLeft, int offsetTop, int childWidth, int childHeight);
+    void copy(WindowInformation *winInfo);
+  };
+
+  class RenderNativeWindow : public ecolab::CairoSurface
   {
   public:
-    void renderToNativeWindow(unsigned long window);
+    /* Perhaps needs to be public for some auto-binding code */
```

Comment:
Definitely not. If you're managing raw pointers in this way (new in constructor/delete in destructor), then the actual pointer *must* be private, and the copy operations *must* be deleted. But most classes in Minsky need to be DCAS (default constructible, copy constructible, assignable and serialisable), so you won't get very far with this strategy.

If you do need pointers (I'm not sure you really need them here, mind you), then using smart pointers is the smart thing to do, as all these issues have been thought out for you. For example, std::shared_ptr is DCAS, and std::unique_ptr is nearly so.